tiledb/sm/group/group.h

```diff
@@ -331,31 +341,26 @@ class Group {
   /**
    * Get count of members
    *
-   * @return tuple of Status and optional member count
+   * @return member count
    */
-  tuple<Status, optional<uint64_t>> member_count() const;
+  uint64_t member_count() const;
 
   /**
    * Get a member by index
    *
    * @param index of member
-   * @return Tuple of Status, URI string, ObjectType, optional name
+   * @return Tuple of URI string, ObjectType, name
    */
-  tuple<
-      Status,
-      optional<std::string>,
-      optional<ObjectType>,
-      optional<std::string>>
+  tuple<optional<std::string>, optional<ObjectType>, optional<std::string>>
```

Comment:
Convert to tuple<std::string, ObjectType, std::string>... Optionals are no longer required here I think.

Reply:
Should we keep the third argument as an `optional<std::string>`? This corresponds to the `GroupMember::name_` which is optional.

Reply:
I would be ok with that.